packages/test-runner-intercept/README.md

```diff
@@ -0,0 +1,141 @@
+
+```js
+import { interceptModule } from '@web/test-runner-intercept';
+
+const externalLibrary = await interceptModule('external-library');
+
+// Return the original function that 'external-library' exposed in the `externalFunction` named export
+externalLibrary.externalFunction; // f externalFunction() { ... }
+
+// Rewire the 'external-library' module to return this anonymous function as the `externalFunction` export
+externalLibrary.externalFunction = () => console.log('hello world'); // () => console.log('hello world')
+
+const { externalFunction } = await import('external-library');
+externalFunction; // () => console.log('hello world')
+```
+
+### Caveats
+
+#### Order of imports
+
+In order to intercept a module, the module should not be referenced yet. (Once a module is loaded, the module loader also loads all its dependent modules) This means the module containing the system under test should be loaded _after_ the module interception. As interception is achieved using a function, this also means the system under test should _always_ be loaded using a dynamic import in order to be done after the interception.
+
+```js
+import { interceptModule } from '@web/test-runner-intercept';
+
+// First, intercept
+const externalLibrary = await interceptModule('external-library');
+
+// Second, load the system under test.
+const systemUnderTest = await import('../../src/system-under-test.js');
+
+// Run tests
+...
+```
+
+#### Types of module specifiers
+
+Currently, two types of module specifiers are supported: bare modules and server relative modules.
+
+```javascript
+// bare module, located in `node_modules`
+await interceptModule('external-library');
+
+// server relative module
+await interceptModule('/src/library-to-intercept.js');
+```
+
+In order to use regular relative references, `import.meta.resolve()` and `new URL().pathname` can be used.
+
+```javascript
+// relative module
+await interceptModule(new URL(import.meta.resolve('../src/library-to-intercept.js')).pathname);
+```
+
+## Usage
+
+### Setup
+
+```js
+// web-test-runner.config.mjs
+import { interceptModulePlugin } from '@web/test-runner-intercept/plugin';
+
+export default {
+  plugins: [interceptModulePlugin()],
+};
+```
+
+### Simple test scenario
+
+```js
+// src/getTimeOfDay.js
+import { getCurrentHour } from 'time-library';
+
+export function getTimeOfDay() {
+  const hour = getCurrentHour();
+  if (hour < 6 || hour > 18) {
+    return 'night';
+  }
+  return 'day';
+}
+```
+
+```js
+// test/getTimeOfDay.test.js
+import { interceptModule } from '@web/test-runner-intercept';
+
+const timeLibrary = await interceptModule('time-library');
+const { getTimeOfDay } = await import('../src/getTimeOfDay.js');
```

Comment:
Ok this example clarifies my question above, i'll resolve that after reviewing the rest